### PR TITLE
kubectl book - add `newName` function

### PR DESCRIPTION
**Pull request by pwittrock**
- update the customizing pod template with `newName`

### CODE REVIEW

File: docs/book/pages/app_customization/customizing_pod_templates.md [hunk 1]
```diff
@@ -21,9 +21,14 @@ Common examples include:
...
-in a base by specifying the `images` field in the `kustomization.yaml`.  The `newTag` may
-be specified to override the image tag for images whose image name matches `name`.
+Override the name or tag for an `image` field from a [Pod Template](https://kubernetes.io/docs/concepts/workloads/pods/pod-overview/#pod-templates)
+in a base by specifying the `images` field in the `kustomization.yaml`.
+
+| Field     | Description                                                              | Example Field | Example Result |
+|-----------|--------------------------------------------------------------------------|----------| --- |
+| `name`    | Match images with this image name| `name: nginx`| |
+| `newTag`  | Override the image **tag** for images whose image name matches `name`    | `newTag: new` | `nginx:old` -> `nginx:new` |
+| `newName` | Override the image **name** for images whose image name matches `name`   | `newImage: nginx-special` | `nginx:old` -> `nginx-special:old` |
```
Comment (Liujingfang1) [inline docs/book/pages/app_customization/customizing_pod_templates.md]: Maybe add `digest` as well.

File: docs/book/pages/app_management/container_images.md [hunk 2]
```diff
@@ -25,13 +25,21 @@ the `kustomization.yaml` using the `images` field.  When `images` are
 specified, Apply will override the images whose image name matches `name` with a new
 tag.
 
+
+| Field     | Description                                                              | Example Field | Example Result |
+|-----------|--------------------------------------------------------------------------|----------| --- |
+| `name`    | Match images with this image name| `name: nginx`| |
+| `newTag`  | Override the image **tag** for images whose image name matches `name`    | `newTag: new` | `nginx:old` -> `nginx:new` |
+| `newName` | Override the image **name** for images whose image name matches `name`   | `newImage: nginx-special` | `nginx:old` -> `nginx-special:old` |
```
Comment (Liujingfang1) [inline docs/book/pages/app_management/container_images.md]: Maybe add `digest` as well.

**Comment by pwittrock** [comment]
PTAL

**Pull request merged**